src/main/java/com/example/animalshelter/controller/AnimalController.java

```diff
@@ -0,0 +1,22 @@
+package com.example.animalshelter.controller;
+
+import com.example.animalshelter.model.Animal;
+import com.example.animalshelter.service.AnimalService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController()
+public class AnimalController {
+
+    private final AnimalService animalService;
+
+    public AnimalController(AnimalService animalService) {
+        this.animalService = animalService;
+    }
+
+    @PostMapping(value = "/save", consumes = "application/json", produces = "application/json")
+    public Animal saveAnimal(@RequestBody Animal animal) {
```

Comment:
I would suggest changing the name from `save` to 'add`. What do you think? 
We do not save an animal we add it :)

Reply:
yes, as we talk add will be better choice here, changed.